features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx

```diff
@@ -171,29 +171,33 @@ class LifeCycleUpdate extends Component {
         lcMap.set('Created', 'Create');
         lcMap.set('Retired', 'Retire');
         const isPrototype = api.endpointConfig && api.endpointConfig.implementation_status === 'prototyped';
-        const lifecycleButtons = lifecycleStates
-            .filter(item => item.event !== lcMap.get(lcState.state))
-            .map((item) => {
-                if (item.event === 'Deploy as a Prototype') {
-                    return {
-                        ...item,
-                        disabled: !isPrototype || api.endpointConfig == null,
-                    };
-                }
-                if (item.event === 'Publish') {
-                    return {
-                        ...item,
-                        disabled:
-                            api.endpointConfig == null ||
-                            api.policies.length === 0 ||
-                            api.endpointConfig.implementation_status === 'prototyped',
-                    };
+        const lifecycleButtons = lifecycleStates.map((item) => {
+            const state = { ...item, displayName: item.event };
+            if (state.event === 'Deploy as a Prototype') {
+                return {
+                    ...state,
+                    disabled: !isPrototype || api.endpointConfig == null,
+                };
+            }
+            if (state.event === 'Publish') {
+                let { displayName } = state;
+                if (lcState.state === 'Published') {
+                    displayName = 'Re Deploy';
```

Comment:
```suggestion
                    displayName = 'Redeploy';
```

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/7740/commits/cd50ac4ef19324d9f4ee4f3b6dc01cb590aeb3be